### PR TITLE
Foundation for editing forms filled out while offline

### PR DESCRIPTION
**Pull request by manuelroemer**
Generally enables editing filled out forms while offline.
As mentioned in the dev/sync calls, this will not work for *every* form, but it already covers a good amount of the encounter structure.

### CODE REVIEW

File: src/app/openmrs-api/encounter-resource.service.ts [hunk 1]
```diff
@@ -2,8 +2,9 @@ import { Injectable } from '@angular/core';
 import { HttpClient, HttpParams, HttpHeaders } from '@angular/common/http';
 
 import { of as observableOf, Observable } from 'rxjs';
-import { map, flatMap } from 'rxjs/operators';
+import { map, flatMap, catchError } from 'rxjs/operators';
 import { WindowRef } from '../window-ref';
+import { getOfflineEncounterForForm, generateOfflineUuid } from 'src/offline';
```
Comment (FlorianRappl) [inline src/app/openmrs-api/encounter-resource.service.ts:7]: You mean `../../offline`?

**Review by FlorianRappl — APPROVED** [review]
LGTM :beers:!

File: src/app/openmrs-api/encounter-resource.service.ts (hunk 1 above)
Comment (manuelroemer) [reply src/app/openmrs-api/encounter-resource.service.ts]: Yes, fixed!

**Review by FlorianRappl — APPROVED** [review]
LGTM - thanks!

**Pull request merged**